lib/SARIF/Serializer.h

```diff
@@ -0,0 +1,27 @@
+//
+// Created by Daniel Dominguez on 11/1/23.
+//
+
+#ifndef VANGUARD_SARIFSERIALIZER_H
+#define VANGUARD_SARIFSERIALIZER_H
+
+#include "SARIF.h"
+
+
+namespace vanguard {
+
+    template<typename OutFmt>
+    class SARIF::Serializer {
+    public:
+        virtual OutFmt serialize(SARIF::Message &message) = 0;
+        virtual OutFmt serialize(SARIF::PhysLocation &physLocation) = 0;
+        virtual OutFmt serialize(SARIF::LogicLocation &logicLocation) = 0;
+        virtual OutFmt serialize(SARIF::Location &location) = 0;
+        virtual OutFmt serialize(SARIF::Result &result) = 0;
+        virtual OutFmt serialize(SARIF::Run &run) = 0;
+        virtual OutFmt serialize(SARIF::Log &log) = 0;
```

Comment:
I would make the parameters `const` because a reasonable expectation is that serialization will not modify the input data structures.
```suggestion
        virtual OutFmt serialize(const SARIF::Message &message) = 0;
        virtual OutFmt serialize(const SARIF::PhysLocation &physLocation) = 0;
        virtual OutFmt serialize(const SARIF::LogicLocation &logicLocation) = 0;
        virtual OutFmt serialize(const SARIF::Location &location) = 0;
        virtual OutFmt serialize(const SARIF::Result &result) = 0;
        virtual OutFmt serialize(const SARIF::Run &run) = 0;
        virtual OutFmt serialize(const SARIF::Log &log) = 0;
```